flume/src/test/java/org/apache/apex/malhar/flume/sink/FlumeSinkTest.java

```diff
@@ -79,7 +75,7 @@ public synchronized void advertise(Service<byte[]> service)
       {
         logger.info("Discover invoked");
         try {
-          wait();
+          countdown.await();
```

Comment:
How about a timeout for await, to avoid the same scenario from occurring again

Reply:
The reason for the issue was not that there wasn't a timeout, but incorrect synchronization logic that leads to missed update. I added the timeout nevertheless, to ensure termination even when logic is incorrect.

Reply:
Yes I realized that too from your changes, just for safety.